packages/caravan-fees/src/types.ts

```diff
@@ -0,0 +1,444 @@
+  /**
+   * The value of the output in satoshis.
+   * This is the amount of bitcoin contained in this output.
+   */
+  value: number;
+
+  /**
+   * The scriptPubKey in hexadecimal format.
+   * This script defines the conditions that must be met to spend this output.
+   */
+  scriptPubKey: string;
+
+  /**
+   * The Bitcoin address associated with this output.
+   * This is derived from the scriptPubKey and provides a human-readable format.
+   */
+  address: string;
+
+  /**
+   * Indicates whether this output is spendable by the user.
+   *
+   * This is crucial for CPFP (Child-Pays-for-Parent) operations:
+   * - If true, this output can be used as an input in a child transaction for CPFP.
+   * - This includes both change outputs and any other outputs sent to addresses
+   *   controlled by the user's wallet.
+   *
+   * Note: The term "isChange" is replaced with "isSpendable" to more accurately
+   * reflect its purpose in CPFP scenarios.
+   */
+  isSpendable: boolean;
+}
+
+/**
+ * Represents a fee rate in satoshis per virtual byte.
+ * This is used for fee estimation and fee bumping calculations.
+ * @see https://bitcoinops.org/en/topics/fee-estimation/
+ */
+export type FeeRateSatsPerVByte = number;
+
+/**
+ * Represents an amount in satoshis.
+ * Satoshis are the smallest unit of bitcoin (1 BTC = 100,000,000 satoshis).
+ */
+export type Satoshis = string;
+
+/**
+ * Represents an amount in bitcoin.
+ */
+export type BTC = string;
+
+/**
+ * Configuration options for creating a transaction template.
+ * This is used to set up the initial state and parameters for a new transaction.
+ */
+export interface TransactionTemplateOptions {
+  /**
+   * The target fee rate in satoshis per virtual byte.
+   * This is used to calculate the appropriate fee for the transaction.
+   */
+  targetFeeRate: number;
+
+  /**
+   * The dust threshold in satoshis.
+   * Outputs below this value are considered uneconomical to spend.
+   * @see https://github.com/bitcoin/bitcoin/blob/master/src/policy/policy.cpp
+   */
+  dustThreshold?: Satoshis;
+
+  /**
+   * The Bitcoin network to use (mainnet, testnet, or regtest).
+   */
+  network: Network;
+
+  /**
+   * The type of script used for the transaction (e.g., "p2pkh", "p2sh", "p2wpkh", "p2wsh").
+   * This affects how the transaction is constructed and signed.
+   */
+  scriptType: string;
+
+  /**
+   * Optional array of input templates to use in the transaction.
+   */
+  inputs?: BtcTxInputTemplate[];
+
+  /**
+   * Optional array of output templates to use in the transaction.
+   */
+  outputs?: BtcTxOutputTemplate[];
+
+  /**
+   * The number of signatures required in a multisig setup.
+   * This is used for multisig transactions and affects the transaction size.
+   */
+  requiredSigners?: number;
+
+  /**
+   * The total number of signers in a multisig setup.
+   * This is used along with requiredSigners for multisig transactions.
+   */
+  totalSigners?: number;
+}
+
+export enum ScriptType {
```

Comment:
so I see that this supports more options than our multisig ones. Maybe extend from that rather than create a new one and match the casing and other patterns. This could ensure consistency. 

And if you do have a type like this, you shouldn't be typing address/script types as `string`.

Reply:
so it turns out we cannot extend enums in ts .. so I found a great solution [here](https://stackoverflow.com/questions/52857698/extending-enum-in-typescript)

had to figure out how this would work but ... I have coded it as follows and it looks good to me ... your thoughts ?

```js
/**
 * Enum representing single-signature Bitcoin script types.
 * @enum {string}
 */
export enum SingleSigScriptType {
  P2PKH = "P2PKH",
  P2WPKH = "P2WPKH",
}

/**
 * Enum representing unknown or unsupported script types.
 * @enum {string}
 */
export enum OtherScriptType {
  UNKNOWN = "UNKNOWN",
}

/**
 * Type representing all possible Bitcoin script types.
 * This includes single-signature types, multisignature types from MULTISIG_ADDRESS_TYPES,
 * and an UNKNOWN type for error handling.
 *
 * @typedef {MULTISIG_ADDRESS_TYPES | SingleSigScriptType | OtherScriptType} ScriptType
 */
export type ScriptType =
  | keyof typeof MULTISIG_ADDRESS_TYPES
  | SingleSigScriptType
  | OtherScriptType;

/**
 * Constant object containing all valid ScriptType values.
 * This can be used similarly to an enum, providing easy access to all script types.
 *
 * @constant
 * @type {Record<ScriptType, ScriptType>}
 *
 * @example
 * const scriptType = ScriptType.P2WSH;
 * if (scriptType === ScriptType.P2WSH) {
 *   console.log("This is a native SegWit multisig address");
 * }
 */
export const ScriptType = {
  ...MULTISIG_ADDRESS_TYPES,
  ...SingleSigScriptType,
  ...OtherScriptType,
};
```

Reply:
done